### PR TITLE
Ae upgrade boot2docker

### PR DESCRIPTION
**Pull request by paetling**
review @thieman

### CODE REVIEW

File: docs/changelog.md [hunk 1]
```diff
@@ -1,6 +1,8 @@
 # Changelog
 
 ## 0.1.3 (In Progress)
+  * Add support for boot2docker 1.7. This means supporting a boot2docker vm that runs 64 bit tiny core linux
```
Comment (thieman) [inline docs/changelog.md]: how about `Added support for boot2docker 1.7+, which uses 64-bit Tiny Core Linux`

Comment (paetling) [inline docs/changelog.md]: cool

File: dusty/systems/docker/__init__.py [hunk 2]
```diff
@@ -23,14 +23,26 @@ def get_dusty_images():
...
+    if 'DOCKER_CERT_PATH' in environment_variables:
+        env['DOCKER_CERT_PATH'] = environment_variables['DOCKER_CERT_PATH']
+    if 'DOCKER_TLS_VERIFY' in environment_variables:
+        env['DOCKER_TLS_VERIFY'] = environment_variables['DOCKER_TLS_VERIFY']
+    return env
+
+def get_docker_env():
+    env = _get_set_envs()
+    if len(env.keys()) < 3:
+        output = check_output_demoted(['boot2docker', 'shellinit'])
```
Comment (thieman) [inline dusty/systems/docker/__init__.py]: you killed my stderr thing

File: dusty/systems/docker/__init__.py [hunk 3]
```diff
@@ -23,14 +23,26 @@ def get_dusty_images():
...
-            continue
-        k, v = line.strip().split()[1].split('=')
-        env[k] = v
+    environment_variables = os.environ
+    if 'DOCKER_HOST' in environment_variables:
+        env['DOCKER_HOST'] = environment_variables['DOCKER_HOST']
+    if 'DOCKER_CERT_PATH' in environment_variables:
+        env['DOCKER_CERT_PATH'] = environment_variables['DOCKER_CERT_PATH']
+    if 'DOCKER_TLS_VERIFY' in environment_variables:
+        env['DOCKER_TLS_VERIFY'] = environment_variables['DOCKER_TLS_VERIFY']
```
Comment (thieman) [inline dusty/systems/docker/__init__.py]: can you loop this logic over a list of keys

File: dusty/systems/docker/__init__.py [hunk 4]
```diff
@@ -23,14 +23,26 @@ def get_dusty_images():
...
-def get_docker_env():
-    output = check_output_demoted(['boot2docker', 'shellinit'], redirect_stderr=True)
+def _get_set_envs():
     env = {}
-    for line in output.splitlines():
-        if not line.strip().startswith('export'):
-            continue
-        k, v = line.strip().split()[1].split('=')
-        env[k] = v
+    environment_variables = os.environ
```
Comment (thieman) [inline dusty/systems/docker/__init__.py]: Why do we need to do this? More verbose, level of indirection for no gain

File: dusty/systems/docker/__init__.py (hunk 2 above)
Comment (paetling) [inline dusty/systems/docker/__init__.py]: no i did not

Comment (paetling) [inline dusty/systems/docker/__init__.py]: DAMN YOU TRAVIS

File: dusty/systems/virtualbox/__init__.py [hunk 5]
```diff
@@ -9,7 +9,12 @@
 
 def _ensure_rsync_is_installed():
     logging.info('Installing rsync inside the Docker VM')
-    check_and_log_output_and_error_demoted(['boot2docker', 'ssh', 'tce-load -wi rsync'])
+    version = check_output_demoted(['boot2docker', 'ssh', 'uname -m'])
+    if '_64' in version:
+        # rsync 64 bit binary does not exist for tiny core linux. Made our own binary and are pulling and installing that
+        check_and_log_output_and_error_demoted(['boot2docker', 'ssh', 'curl https://64bit-rsync.s3.amazonaws.com/rsync > rsync; sudo chmod 755 rsync; sudo mv rsync /usr/bin/rsync'])
+    else:
+        check_and_log_output_and_error_demoted(['boot2docker', 'ssh', 'tce-load -wi rsync'])
```
Comment (thieman) [inline dusty/systems/virtualbox/__init__.py]: Once this is merged and good let's make sure to upgrade the Mac slave fully and make sure everything is still legit

File: dusty/systems/virtualbox/__init__.py [hunk 6]
```diff
@@ -9,7 +9,12 @@
 
 def _ensure_rsync_is_installed():
     logging.info('Installing rsync inside the Docker VM')
-    check_and_log_output_and_error_demoted(['boot2docker', 'ssh', 'tce-load -wi rsync'])
+    version = check_output_demoted(['boot2docker', 'ssh', 'uname -m'])
+    if '_64' in version:
+        # rsync 64 bit binary does not exist for tiny core linux. Made our own binary and are pulling and installing that
+        check_and_log_output_and_error_demoted(['boot2docker', 'ssh', 'test -f /usr/bin/rsync || curl https://64bit-rsync.s3.amazonaws.com/rsync > rsync; sudo chmod 755 rsync; sudo mv rsync /usr/bin/rsync'])
+    else:
+        check_and_log_output_and_error_demoted(['boot2docker', 'ssh', 'tce-load -wi rsync'])
```
Comment (thieman) [inline dusty/systems/virtualbox/__init__.py]: We should just check whether `rsync` is in the path before doing any of this stuff. `which rsync` or whatever

**Pull request merged**